src/praisonai-agents/praisonaiagents/tools/edit_tools.py

```
@@ -116,7 +140,105 @@ def _render_diff(self, old_content: str, new_content: str, filepath: str, max_li
+        """
+        if self._post_edit_diagnostics == "off":
+            return ""
+        try:
+            resolved = self._diagnostics_command(safe_path)
+            if resolved is None:
+                return ""
+            tool_name, argv = resolved
+
+            import subprocess
+            try:
+                proc = subprocess.run(
```

Ruff (0.15.18)

[error] `subprocess` call: check for execution of untrusted input

(S603)